### PR TITLE
Fix nasty bug in UDPSocket::close().

### PR DESCRIPTION
**Pull request by f1rmb**

### CODE REVIEW

File: UDPSocket.cpp [hunk 1]
```diff
@@ -188,6 +188,8 @@ bool CUDPSocket::open(const unsigned int index, const unsigned int af, const std
 		return false;
 	}
 
+	close(index);
```
Comment (shawnchain) [inline UDPSocket.cpp:191]: defensive code

File: UDPSocket.cpp [hunk 2]
```diff
@@ -344,13 +346,13 @@ bool CUDPSocket::write(const unsigned char* buffer, unsigned int length, const s
 
 void CUDPSocket::close()
 {
-	for (int i = 0; i < UDP_SOCKET_MAX; i++)
-		close(m_fd[i]);
+	for (unsigned int i = 0; i < UDP_SOCKET_MAX; i++)
+		close(i);
```
Comment (shawnchain) [inline UDPSocket.cpp:350]: key to to the fix

File: UDPSocket.cpp (hunk 1 above)
Comment (f1rmb) [reply UDPSocket.cpp:191]: At least the previous opened socket if any, is closed, and it marked at -1 in the array.

File: UDPSocket.cpp (hunk 2 above)
Comment (f1rmb) [reply UDPSocket.cpp:350]: Yes, instead if passing file descriptor as an array index (and corrupt memory, like timers in DMRGateay's CDMRNetwork class, messing the timeout values).

**Pull request merged**